strings/base_activation.h

```diff
@@ -400,10 +400,10 @@ namespace winrt::impl
         return factory.call(static_cast<CastType>(callback));
     }
 
-    template <typename Class, typename Interface = Windows::Foundation::IActivationFactory>
-    com_ref<Interface> try_get_activation_factory(hresult_error* exception = nullptr) noexcept
+    template <typename Interface = Windows::Foundation::IActivationFactory>
+    com_ref<Interface> try_get_activation_factory(std::wstring_view className, hresult_error* exception = nullptr) noexcept
     {
-        param::hstring const name{ name_of<Class>() };
+        param::hstring const name{ className };
```

Comment:
This named temporary isn't needed - may just want to pass the `className` directly to the `get_runtime_activation_factory` function as it can create the `param::hstring` automatically.